tools/cli/lib/api.c

```diff
@@ -671,6 +671,322 @@ TDNFCliRepoSyncCommand(
+    char **ppszLines = NULL;
+    char *pszdependencies=NULL;
+    PTDNF_PKG_INFO ppPkgInfo = (PTDNF_PKG_INFO)data;
+
+    if(!pContext || !pContext->hTdnf || !pCmdArgs || !pContext->pFnRepoQuery)
+    {
+        dwError = ERROR_TDNF_CLI_INVALID_ARGUMENT;
+        BAIL_ON_CLI_ERROR(dwError);
+    }
+
+    dwError = TDNFCliParseRepoQueryArgs(pCmdArgs, &pRepoqueryArgs);
+    BAIL_ON_CLI_ERROR(dwError);
+
+    dwError = pContext->pFnRepoQuery(pContext, pRepoqueryArgs, &pPkgInfos, &dwCount);
+    BAIL_ON_CLI_ERROR(dwError);
+
+    for (i = 0; i < (int)dwCount; i++)
+      {
+         pPkgInfo = &pPkgInfos[i];
+
+         if (pPkgInfo->ppszDependencies)
+         {
+             for (j = 0; pPkgInfo->ppszDependencies[j]; j++);
+             nCount += j;
+         }
+         else if (pPkgInfo->ppszFileList)
+         {
+             for (j = 0; pPkgInfo->ppszFileList[j]; j++);
+             nCount += j;
+         }
+      }
+      if (nCount > 0)
+      {
+          dwError = TDNFAllocateMemory(nCount + 1, sizeof(char *), (void**)&ppszLines);
+          BAIL_ON_CLI_ERROR(dwError);
+          for (k = 0, i = 0; i < (int)dwCount; i++)
+          {
+              pPkgInfo = &pPkgInfos[i];
+
+              if (pPkgInfo->ppszDependencies)
+              {
+                  for (j = 0; pPkgInfo->ppszDependencies[j]; j++)
+                  {
+                      ppszLines[k++] = pPkgInfo->ppszDependencies[j];
+                  }
+              }
+              else if (pPkgInfo->ppszFileList)
+              {
+                  for (j = 0; pPkgInfo->ppszFileList[j]; j++)
+                  {
+                      ppszLines[k++] = pPkgInfo->ppszFileList[j];
+                  }
+              }
+          }
+
+          dwError = TDNFStringArraySort(ppszLines);
+          BAIL_ON_CLI_ERROR(dwError);
+
+          nSepSize = strlen("\n");
+          nSize = nSepSize * (nCount + 1);
+          for(i = 0; ppszLines[i]; i++)
+          {
+              if(i == 0 || strcmp(ppszLines[i], ppszLines[i-1]))
+                nSize += strlen(ppszLines[i]);
+          }
+          nSize++;
+
+          dwError = TDNFAllocateMemory(nSize, sizeof(char), (void**)&pszdependencies);
+          BAIL_ON_CLI_ERROR(dwError);
+
+          dwError = TDNFAllocateMemory(1, sizeof(char *), (void**)&ppPkgInfo->ppszDependencies);
+          BAIL_ON_CLI_ERROR(dwError);
+
+          for (j = 0; ppszLines[j]; j++)
+          {
+              if (j == 0)
+              {
+                   strcpy(&pszdependencies[p], ppszLines[j]);
+                   p += strlen(ppszLines[j]);
+              } else if (strcmp(ppszLines[j], ppszLines[j-1]))
+              {
+                   strcpy(&pszdependencies[p], "\n");
+                   p += nSepSize;
+                   strcpy(&pszdependencies[p], ppszLines[j]);
+                   p += strlen(ppszLines[j]);
+              }
+          }
+
+          dwError = TDNFAllocateString(pszdependencies, &ppPkgInfo->ppszDependencies[0]);
```

Comment:
So basically, you are concatenating all string elements into one. We shouldn't do that, it's unmaintainable.